src/dtx/dtx_resync.c

```diff
@@ -565,6 +606,18 @@ dtx_resync(daos_handle_t po_hdl, uuid_t po_uuid, uuid_t co_uuid, uint32_t ver,
 		return rc;
 	}
 
+	crt_group_rank(NULL, &myrank);
+
+	pool = cont->sc_pool->spc_pool;
+	ABT_rwlock_rdlock(pool->sp_lock);
+	rc = pool_map_find_target_by_rank_idx(pool->sp_map, myrank,
+					      dss_get_module_info()->dmi_tgt_id, &target);
+	D_ASSERT(rc == 1);
+	ABT_rwlock_unlock(pool->sp_lock);
+
+	if (target->ta_comp.co_status == PO_COMP_ST_UP)
+		dra.for_discard = 1;
+
```

Comment:
I thought only check during pool map update.  unless dtx_resync() will only be called during pool map update?

Reply:
Currently, the dtx_resync will be triggered for any pool map changes, for example, one rebuild action will trigger dtx_resync twice: one is just before the rebuild start, the other is when rebuild done with pool map refreshed. In theory, the 2nd dtx_resync is unnecessary. But unfortunately, our current pool map refresh mechanism does not contain enough knowledge to distinguish the two cases. So dtx_resync has to depends on itself logic to guarantee the correctness even if be called repeatedly. Here when the dtx_resync is called for the 2nd time, related target's status will be "UPIN", so discard will be done only once.